src/LiveComponent/tests/Functional/Form/ComponentWithFormTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\UX\LiveComponent\Tests\Functional\EventListener;
+
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\DomCrawler\Crawler;
+use Symfony\UX\LiveComponent\LiveComponentHydrator;
+use Symfony\UX\TwigComponent\ComponentFactory;
+use Zenstruck\Browser\Response\HtmlResponse;
+use Zenstruck\Browser\Test\HasBrowser;
+use Zenstruck\Foundry\Test\Factories;
+use Zenstruck\Foundry\Test\ResetDatabase;
+
+/**
+ * @author Jakub Caban <kuba.iluvatar@gmail.com>
+ */
+class ComponentWithFormTest extends KernelTestCase
+{
+    use Factories;
+    use HasBrowser;
+    use ResetDatabase;
+
+    public function testFormValuesRebuildAfterFormChanges(): void
+    {
+        /** @var LiveComponentHydrator $hydrator */
+        $hydrator = self::getContainer()->get('ux.live_component.component_hydrator');
+        /** @var ComponentFactory $factory */
+        $factory = self::getContainer()->get('ux.twig_component.component_factory');
+        $component = $factory->create('form_with_collection_type');
+
+        $dehydrated = $hydrator->dehydrate($component);
+        $token = null;
+
+        $this->browser()
+            ->get('/_components/form_with_collection_type?'.http_build_query($dehydrated))
+            ->use(function (HtmlResponse $response) use (&$dehydrated, &$token) {
+                // mimic user typing
+                $dehydrated['blog_post_form']['content'] = 'changed description by user';
+                $dehydrated['validatedFields'] = ['blog_post_form.content'];
+                $token = $response->crawler()->filter('div')->first()->attr('data-live-csrf-value');
+            })
+            // post to action, which will change the "name" field in the form
+            ->post('/_components/form_with_collection_type/addComment?'.http_build_query($dehydrated), [
+                'headers' => ['X-CSRF-TOKEN' => $token],
+            ])
+            ->assertStatus(422)
+            ->dump()
+            // look for original embedded form
```

Comment:
```suggestion
            ->assertStatus(422)
            // look for original embedded form
```

Reply:
Oh, but the dump is such a cool feature! ;)